File: server/medicore.layer/MediCore.Service/Model/HospitalAppointmentDetailModel.cs
using dna.core.service.Models.Abstract;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace MediCore.Service.Model
{
    public class HospitalAppointmentDetailModel : IModelBase
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "Required")]
        public int HospitalAppointmentId { get; set; }

        [Required(ErrorMessage = "Required")]
        public int PatientId { get; set; }

        [StringLength(100, ErrorMessage = "StringLength")]
        public string Problem { get; set; }

        
        public virtual HospitalModel Hospital { get; set; }
       
        public virtual PatientModel Patient { get; set; }

        public virtual HospitalAppointmentModel HospitalAppointment { get; set; }
    }
}

File: smartmedika.admin/src/app/model/hospitalAppointmentDetailModel.ts
import { IEntityBase } from './IEntityBase';

import { HospitalModel } from './hospitalModel';
import { PatientModel} from './patientModel';
import { HospitalAppointmentModel } from './hospitalAppointmentModel';

export class HospitalAppointmentDetailModel implements IEntityBase{
    id: number;
    hospitalAppointmentId: number;
    problem: string;
 
    hospital: HospitalModel;
    patient : PatientModel;    
    hospitalAppointment: HospitalAppointmentModel;    
    

    constructor(id:number, hospitalAppointmentId: number, patient: PatientModel ) {
        this.id = 0;
        this.hospitalAppointmentId = hospitalAppointmentId;
        //this.problem = '';
        this.patient = patient;
    }
}